tests/phpunit/tests/block-template.php

```diff
@@ -275,6 +275,82 @@ public function test_get_the_block_template_html_does_not_generally_enforce_loop
 		$this->assertSame( array( false, true ), $in_the_loop_logs, 'Main query loop was triggered incorrectly' );
 	}
 
+	/**
+	 * @ticket 58319
+	 *
+	 * @covers ::get_block_theme_folders
+	 * @dataProvider data_get_block_theme_folders
+	 */
+	public function test_get_block_theme_folders( $theme, $expected, $message ) {
+		$wp_theme   = wp_get_theme( $theme );
+		$reflection = new ReflectionMethod( $wp_theme, 'cache_get' );
+		$reflection->setAccessible( true );
+		$wp_theme->cache_delete(); // Clear cache.
+
+		$this->assertSame( $expected, get_block_theme_folders( $theme ), $message );
+		$theme_cache  = $reflection->invoke( $wp_theme, 'theme' );
+		$cached_value = $theme_cache['block_template_folders'];
+		$reflection->setAccessible( false );
+
+		$this->assertSame( $expected, $cached_value, 'Cache - ' . $message );
+	}
+
+	/**
+	 * Data provider for `test_get_block_theme_folders()`.
+	 */
+	public function data_get_block_theme_folders() {
+		return array(
+			'block-theme'                       => array(
+				'block-theme',
+				array(
+					'wp_template'      => 'templates',
+					'wp_template_part' => 'parts',
+				),
+				'For block-theme, expected `templates` and `parts` and `wp_template` and `wp_template_part` respectively',
```

Comment:
```suggestion
				'For block-theme, expected `wp_template` to be `templates` and `wp_template_part` to be `parts`.',
```
IMO, this reads a little clearer. What do you think?

:1234: Applies to the other datasets.

Reply:
Updated @costdev.